From 4a1ad05f26819427e633ef8fda40df2f172c7221 Mon Sep 17 00:00:00 2001
From: Dinolek <git@dinolek.me>
Date: Sun, 6 Jul 2025 22:46:32 +0200
Subject: [PATCH 006/115] arm64: dts: mediatek: Add basic support for Xiaomi
 Redmi Note 8 Pro

This smartphone uses a MediaTek MT6785 system-on-chip with 4G-8G of RAM.
It can currently boot into initramfs with u-boot.

Signed-off-by: Dinolek <git@dinolek.me>
---
 arch/arm64/boot/dts/mediatek/Makefile               |  1 +
 .../boot/dts/mediatek/mt6785-xiaomi-begonia.dts     | 13 +++++++++++++
 2 files changed, 14 insertions(+)
 create mode 100644 arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia.dts

diff --git a/arch/arm64/boot/dts/mediatek/Makefile b/arch/arm64/boot/dts/mediatek/Makefile
index a86fb313b..0523b04ae 100644
--- a/arch/arm64/boot/dts/mediatek/Makefile
+++ b/arch/arm64/boot/dts/mediatek/Makefile
@@ -2,6 +2,7 @@
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt2712-evb.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt6755-evb.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt6779-evb.dtb
+dtb-$(CONFIG_ARCH_MEDIATEK) += mt6785-xiaomi-begonia.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt6795-evb.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt6795-sony-xperia-m5.dtb
 dtb-$(CONFIG_ARCH_MEDIATEK) += mt6797-evb.dtb
diff --git a/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia.dts b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia.dts
new file mode 100644
index 000000000..120761db1
--- /dev/null
+++ b/arch/arm64/boot/dts/mediatek/mt6785-xiaomi-begonia.dts
@@ -0,0 +1,13 @@
+// SPDX-License-Identifier: (GPL-2.0 OR MIT)
+/*
+ * Copyright (c) 2025 Dinolek <git@dinolek.me>
+ */
+
+/dts-v1/;
+#include "mt6785.dtsi"
+
+/ {
+	model = "Xiaomi Redmi Note 8 Pro";
+	compatible = "xiaomi,begonia", "mediatek,mt6785";
+	chassis-type = "handset";
+};
-- 
2.56.0

